numpy/linalg/linalg.py

```diff
@@ -2811,3 +2811,320 @@ def _multi_dot(arrays, order, i, j, out=None):
+    where lambda is as estimated by the just described process.
+    
+    Calls: 
+           x = autoreg(A,b)
+           x, ur, sigma, lambda = autoreg2(A,b)     , for details
+           x = autoregnn(A,b)    , to force the solution to be non-negative
+    ----------
+    A : (M, N) array_like "Coefficient" matrix.
+    b : (M) 1-D array_like Ordinate or "dependent variable" values. 
+
+    Returns
+    -------
+    x : (N) ndarray
+        Least-squares solution, with minimal residual, if the system is well behaved.
+        Regularized solution, with larger residual, if the system is badly conditioned.
+        Regularized and non-negative solution if solvenn() is called, possibly with yet
+        larger residual.
+    ur: the "usable rank" : int
+    sigma : estimate Right hand Side RMS error: float
+    lambda : the estimated Tikhonov regularization parameter: float
+
+    Raises
+    ------
+    Autoreg is believed to be protected from virtually all calculation faults.
+    However, Autoreg calls this package's SVD algorithm, which may raise this:
+    
+    LinAlgError:
+        If computation does not converge.
+
+    Examples
+    --------
+    See documentation of lstsq in this package for a standard least-square example.
+    These routines will behave the same as lstsq when the system is well behaved.
+    Here is a tiny example of an ill-conditioned system as handled by autoreg.
+
+    x + y = 2
+    x + 1.01 y =3
+    
+    Then A = array([[ 1.,  1.],
+                    [ 1.,  1.01.]])
+    and b = array([2.0, 3.0])
+    
+    Then standard solvers like lstsq will return:
+    x = [-98. , 100.]
+    
+    But autoreg.solve(A,b) will recognize the violation of the Picard Condition and return
+    x = [1.12216 , 1.12779]
+
+    Beware that the results of any equation solver, and especially this one, should always
+    be checked for appropriateness for your problem. Especially check the residual by
+    examining A*x - b.
+    
+    For more details and examples see http://www.rejones7.net/autorej/ .
+    
+    References
+    
+    Re Picard Condition: "The discrete picard condition for discrete ill-posed problems", 
+    Per Christian Chansen, 1990     https://link.springer.com/article/10.1007/BF01933214
+    
+    Re Nonnegative solutions: "Solving Least Squares Problems", by Charles L. Lawson 
+    and Richard J. Hanson. Prentice-Hall 1974
+    
+    Re autoreg algorithms: "Solving Linear Algebraic Systems Arising in the Solution of 
+    Integral Equations of the First Kind", Dissertation by Rondall E. Jones, 1985, U. of N.M.
+    Advisor: Cleve B. Moler, creator of MatLab and co-founder of MathWorks.
+
+"""
+from math import sqrt
+import numpy as np
+import sys
+
+# Compute two-norm of x[:] with no overflow even if values are very large.
+# If x=[1.0 , 2.0 , 1.0E200, 5.0], then numpy.linalg.norm(x)
+# incorrectly gives "inf" as the answer.
+# Two_norm(x) correctly gives 1.0E200 as the answer.
```

Comment:
You should open an issue about this - I think we should fix linalg.norm

Reply:
FYI: `scipy.linalg.norm([1.0 , 2.0 , 1.0E200, 5.0])` gives `1e+200`.

Reply:
Interesting.
I was trying to restrict autoreg to using numpy libraries only. 
Since no one has brought up this overflow problem before I think numpy's norm could be left as is so as not to increase execution time for general users. But it is easy to repair, as you see in autoreg, if the community wants to repair it.
Ron

Reply:
@WarrenWeckesser: Perhaps then we should document the speed / precision tradeoff, if we decide not to fix it.